LandLord-latest/src/main/java/biz/princeps/landlord/LandLord.java

```diff
@@ -60,6 +60,17 @@ public void onEnable() {
 
         new PistonOverwriter(this);
 
+        // Handle 1.18 auto-migration.
+        // https://minecraft.fandom.com/wiki/Data_version
+        int currentDataVersion = getServer().getUnsafe().getDataVersion();
+        if (currentDataVersion > 2858 && getConfig().getBoolean("check-1-18-world-heights")) {
+            getLogger().warning("It appears that you are running Minecraft 1.18 for the first time. " +
+                    "Due to world height changes in overworld, Landlord will try to convert heights in configuration and on lands. " +
+                    "This operation could be unsuccessful, so don't hesitate to check configuration/regions, contact us and use the command '/ll update -c'!");
+            getConfig().set("check-1-18-world-heights", false);
+            saveConfig();
+            getServer().dispatchCommand(getServer().getConsoleSender(), "ll update -c");
```

Comment:
This should not be done automatically in my opinion.

Reply:
That makes sense.